mono/utils/mono-dl.h

```diff
@@ -29,11 +29,11 @@ typedef struct {
 } MonoDl;
 
 
-MONO_API MonoDl*     mono_dl_open       (const char *name, int flags, char **error_msg) MONO_LLVM_INTERNAL;
+MONO_API MonoDl* mono_dl_open  (const char *name, int flags, char **error_msg) MONO_LLVM_INTERNAL_NO_EXTERN_C;
 MONO_EXTERN_C
-char*       mono_dl_symbol     (MonoDl *module, const char *name, void **symbol) MONO_LLVM_INTERNAL;
+char*       mono_dl_symbol     (MonoDl *module, const char *name, void **symbol) MONO_LLVM_INTERNAL_NO_EXTERN_C;
 MONO_EXTERN_C
-void        mono_dl_close      (MonoDl *module) MONO_LLVM_INTERNAL;
+void        mono_dl_close      (MonoDl *module) MONO_LLVM_INTERNAL_NO_EXTERN_C;
```

Comment:
These are surprising.

Reply:
Oh wait, no they're not, I guess. It's because we want them `extern "C"` always, even if we're not doing loaded LLVM.

Reply:
Right. It is a delicate dance among related but not contradictory factors:

 - functions
 - data
 -MONO_API
 - LOADED_LLVM being conditional. These could also be moved to start of line I think, and maybe mildly preferable, for consistency, and to give LOADED_LLVM on Windows a chance, if that ever matters.